app/gather/api/utils.py

```diff
@@ -0,0 +1,252 @@
+# Copyright (C) 2020 by eHealth Africa : http://www.eHealthAfrica.org
+#
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import logging
+import os
+from requests.exceptions import HTTPError
+from aether.python.utils import request
+
+LOG = logging.getLogger('Utils')
+
+
+def consumer_request(url='', method='get', data=None, headers={}):
+    '''
+    Handle request calls to a consumer
+    '''
+    headers['content-type'] = 'application/json'
+    res = request(
+        method=method,
+        url=url,
+        json=data or {},
+        headers=headers,
+    )
+    try:
+        res.raise_for_status()
+    except HTTPError as e:
+        LOG.debug(str(e))
+        raise e
+    try:
+        return res.json()
+    except Exception:
+        return res.content
+
+
+def upsert_resource(base_url, resource_name, resource, headers={}, skip_subs=False):
+    '''
+    Upsert consumer resources
+    '''
+    _resource_id = resource.get('id')
+    if _resource_id:
+        try:
+            _existing_resource = consumer_request(
+                url=f'{base_url}{resource_name}/list',
+                headers=headers
+            )
+            _method = f'update?id={_resource_id}' if _resource_id in _existing_resource else 'add'
+            url = f'{base_url}{resource_name}/{_method}'
+
+            if resource_name == 'job' and _method != 'add' and not skip_subs:
+                _existing_job = consumer_request(
+                    url=f'{base_url}job/get?id={_resource_id}',
+                    headers=headers
+                )
+                old_subs = _existing_job.get('subscription', [])
+                new_subs = list(set(old_subs + resource['subscription']))
+                resource['subscription'] = new_subs
+
+            return consumer_request(
+                url=url,
+                method='post',
+                headers=headers,
+                data=resource
+            )
+        except Exception as e:
+            LOG.debug(str(e))
+            raise e
+    else:
+        err = f'{resource_name}: No id provided'
+        LOG.debug(err)
+        raise Exception(err)
+
+
+def __clean_name(value):
```

Comment:
the point of using `__` is to indicate that the method is private and not use it outside the module. Looking at how you are using it maybe it's better to use it within the method `configure_consumers` something like:
```python
def configure_consumers(consumer_settings, survey_name, headers={}):
    # ...
    survey_name = __clean_name(survey_name)
```